packages/mobile/src/identity/contactMapping.ts

```diff
@@ -39,13 +43,37 @@ import { getConnectedAccount } from 'src/web3/saga'
 import { currentAccountSelector } from 'src/web3/selectors'
 
 const TAG = 'identity/contactMapping'
+export const IMPORT_CONTACTS_TIMEOUT = 2 * 60 * 1000 // 2 minutes
+
+export enum ImportContactsStatus {
+  Failed = -1,
+  Stopped = 0,
+  Prepping = 1,
+  Importing = 2,
+  Processing = 3,
+  Matchmaking = 4,
+  Done = 5,
+}
 
-export function* doImportContactsWrapper() {
+export function* doImportContactsWrapper({ doMatchmaking }: ImportContactsAction) {
   yield call(getConnectedAccount)
   try {
     Logger.debug(TAG, 'Importing user contacts')
 
-    yield call(doImportContacts)
+    const { result, cancel, timeout } = yield race({
```

Comment:
Nice implementation on this

Reply:
Yeah this is a Saga pattern we use in a few places, like verification too